### PR TITLE
feat:-Sorting and filter

### PR DESCRIPTION
**Pull request by shubhamkumar648**
## file to check:

1.src/component/filter/Filter.js
2.src/contexts/Filter-context.js

## helper function 
1. src/utils/filterbyTag.js
2. src/utils/sortByDate.js
3. src/utils/filterbyPriority.js
chrome-capture (4)

# note:- avoid css i  will do later

### CODE REVIEW

File: src/pages/note/Note.js [hunk 1]
```diff
@@ -12,13 +13,13 @@ export const Note = () => {
   const { notesState } = useNotes();
   const { notes } = notesState; 
  const {stateFilter} = useFilter()
-  const {priority,sortBydate} = stateFilter
+  const {priority,sortBydate,tags} = stateFilter
 
 
 
-// const NotePriorityFilter = priorityFilter(priority,notes)
```
Comment (aniketsingh13) [inline src/pages/note/Note.js]: remove commented  code here

File: src/pages/note/Note.js [hunk 2]
```diff
@@ -12,13 +13,13 @@ export const Note = () => {
...
-const SortByDateFilter = sortByDateFun(sortBydate,notes)
-
+const NotePriorityFilter = priorityFilter(priority,notes)
+// const SortByDateFilter = sortByDateFun(sortBydate,notes)
```
Comment (aniketsingh13) [inline src/pages/note/Note.js]: here also

File: src/pages/note/Note.js [hunk 3]
```diff
@@ -29,7 +30,8 @@ const SortByDateFilter = sortByDateFun(sortBydate,notes)
 
           <Notes />
            <div className="Notecard">
-          {SortByDateFilter.map((item) => {
+           {/* {console.log(SortByDateFilter,"sortByDte")} */}
```
Comment (aniketsingh13) [inline src/pages/note/Note.js]: remove console here

File: src/reducer/filterReducer.js [hunk 4]
```diff
@@ -1,11 +1,10 @@
 export const  filterReducerFunction = (state,action) => {
-  console.log(state.type);
-  console.log(action.payload);  
+  // console.log(state.type);
+  // console.log(action.payload);
```
Comment (aniketsingh13) [inline src/reducer/filterReducer.js]: here also

**Review by aniketsingh13 — APPROVED** [review]
code structure look good 
code is also easy to read
add a good video that properly describe your feature

File: src/reducer/filterReducer.js (hunk 4 above)
Comment (shubhamkumar648) [reply src/reducer/filterReducer.js]: thanks i will do

**Comment by shubhamkumar648** [comment]
next time i will do proper video

**Review by vaibhav-mahalle — APPROVED** [review]
remove commented code from before raising PR, everything else looks good

**Pull request merged**